security/cve_2011_4815_spec.rb

```diff
@@ -0,0 +1,35 @@
+require File.expand_path('../../spec_helper', __FILE__)
+
+describe :resists_cve_2011_4815, shared: true do
+  it "resists CVE-2011-4815 by having different hash codes in different processes" do
+    eval("(#{@method}).hash.to_s").should_not == ruby_exe("print (#{@method}).hash")
+  end
+end
+
+describe "Object#hash" do
+  it_behaves_like :resists_cve_2011_4815, 'Object.new'
+end
+
+describe "Integer#hash" do
+  it_behaves_like :resists_cve_2011_4815, '14'
```

Comment:
This would benefit from also checking numbers >= `2^63`.
`bignum_value.to_s` can be used to generate such a number.